cmd/kubeadm/app/kubeadm.go

```diff
@@ -30,6 +32,11 @@ func Run() error {
 	pflag.CommandLine.MarkHidden("google-json-key")
 	pflag.CommandLine.MarkHidden("log-flush-frequency")
 
+	// We want to use for HTTP DefaultTransport same defaults
+	// as API machinery will be using for all other calls.
+	// This is critical for better NO_PROXY handling
+	netutil.SetTransportDefaults(http.DefaultTransport.(*http.Transport))
```

Comment:
Don't mess with the default http transport.  We use this to set values on new transports.

Reply:
That's the point here. Everything that is inside kubeadm using http should be copying same Proxy handler that will support CIDR notation in NO_PROXY.

Reply:
do it on a new transport you create. don't mutate the package default `http.DefaultTransport`

Reply:
and how to guarantee that *all* pieces that kubeadm can potentially call directly or from it's dependencies would be following same NO_PROXY rules ? some dependency can easily call `http.Get()` (as not every single usage creates its own transport) and it will go to wrong direction.  In cases where enterprise networks have "interesting" rules about access to internet and proxies don't have access to intranet, those issues are nightmare to debug and workarround.

Reply:
if that's a concern, I'd suggest sweeping the network usage of kubeadm. mutating the http package var is not appropriate.

Reply:
That will help with current kubeadm code, but will open huge hole to new issues in the future.
What is the biggest issue to apply different defaults to default transport in case of kubeadm, which is standalone binary and doesn't affect anything else outside of it ?

Reply:
@liggitt inside kubeadm, there are few places which accessing nodes by IP address and not by domain name. e.g. RetrieveValidatedClusterInfo() or WaitForHealthyKubelet(). External resources might be less critical, but might be some corner cases.
Normal setup for no_proxy is usually something like "no_proxy=127.0.0.1,localhost,.example.com". Those calls for specific IP will usually go to $http_proxy and eventually fail. My usual daily pain with test cluster is about not to forget to put dynamic IP of master node to no_proxy every time I execute kubeadm, instead of just putting in .profile: "export NO_PROXY=....,10.0.0.0/8", and standard Go http client doesn't work well for that.

Reply:
It's fine to construct the client used to contact nodes with a transport built using this defaulter. Just don't modify the http package variable.